src/main/java/org/openpnp/machine/reference/ReferenceCamera.java

```diff
@@ -178,12 +181,37 @@ protected BufferedImage transformImage(BufferedImage image) {
         
         // apply affine transformations
         if (rotation != 0) {
-            // TODO: Fix cropping of rotated image:
-            // http://stackoverflow.com/questions/22041699/rotate-an-image-without-cropping-in-opencv-in-c
-            Point center = new Point(mat.width() / 2D, mat.height() / 2D);
-            Mat mapMatrix = Imgproc.getRotationMatrix2D(center, rotation, 1.0);
-            Imgproc.warpAffine(mat, mat, mapMatrix, mat.size(), Imgproc.INTER_LINEAR);
-	    mapMatrix.release();
+		double rot = rotation;
```

Comment:
In short. It fixes rotation by 90 degree as commented on issue #228 .
affine rotation was leaved as it could be useful. Size was not calc on
affine rotation as  suggested by you because of fast template match
requirement constraints.
I don't see any utility in giving possibility to have rotation to 30
or 45 degree,  but understand that it could be necessary to rotate up
to 3 degree for removing mechanical misalignments. Because this I have
limited it to 13 degree giving additional 10 deg. Of rotation. If you
prefer I remove this limit. The variable rot is not more then variable
rotation, only that its normalized on positive domain for 90 deg
rotation.
45 degree is equivalent to 0.5 rounding on 90 deg rotation.
I know, the calc is wrong if more then +-360 deg is inserted.

made some tests:

640:480 resolution USB 2.1 Webcam , FPS calc every 100 images.

Rotation 133 degree (transpose+flip+warp-affine)
FPS: 66.403
FPS: 78.78
FPS: 91.117

Rotation 90 degree (transpose+flip)
FPS: 113.052
FPS: 104.015
FPS: 125.325

Rotation 3 degree (warp-affine)
FPS: 135.428
FPS: 147.6

Rotation 0 degree
FPS: 158.309
FPS: 169.418
FPS: 181.11